### PR TITLE
global 6.6.6

### PR DESCRIPTION
**Pull request by khirbat**
Also update resource and Python version

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/global.rb [hunk 1]
```diff
@@ -35,17 +37,20 @@ class Global < Formula
...
-    sha256 "98c8aa5a9f778fcd1026a17361ddaf7330d1b7c62ae97c3bb0ae73e0b9b6b0fe"
+    url "https://files.pythonhosted.org/packages/ba/6e/7a7c13c21d8a4a7f82ccbfe257a045890d4dbf18c023f985f565f97393e3/Pygments-2.9.0.tar.gz"
+    sha256 "a18f47b506a429f6f4b9df81bb02beab9ca21d0a5fee38ed15aef65f0545519f"
   end
 
   def install
     system "sh", "reconf.sh" if build.head?
 
     xy = Language::Python.major_minor_version "python3"
     ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
```
Comment (carlocab) [inline Formula/global.rb:48]:
```suggestion
    ENV.prepend_create_path "PYTHONPATH", libexec/Language::Python.site_packages("python3")
```

File: Formula/global.rb [hunk 2]
```diff
@@ -35,17 +37,20 @@ class Global < Formula
...
-    pygments_args = %W[build install --prefix=#{libexec}]
-    resource("Pygments").stage { system "python3", "setup.py", *pygments_args }
+
+    python3 = Formula["python@#{xy}"].opt_bin/"python3"
+    resource("Pygments").stage do
+      system python3, *Language::Python.setup_install_args(libexec)
+    end
```
Comment (carlocab) [inline Formula/global.rb:53]:
```suggestion
    resource("Pygments").stage do
      system "python3", *Language::Python.setup_install_args(libexec)
    end
```
`xy` comes from querying the `python3` found in `PATH` so we may as well just use the one there directly.

**Review by carlocab — COMMENTED** [review]
Thanks, @khirbat. Just a few suggestions. Please squash these changes with your first commit.

**Comment by carlocab** [comment]
There's an audit failure:
```
==> brew audit global --online --git --skip-style
==> FAILED
global:
  * 'revision 1' should be removed
Error: 1 problem in 1 formula detected
```
and a test failure:
```
Error: global: failed
An exception occurred within a child process:
  MethodDeprecatedError: Calling assert_no_match is deprecated! Use refute_match instead.
```

**Comment by carlocab** [comment]
Fixed up the test for you.

**Comment by khirbat** [comment]
Thanks @carlocab 

I have fixed the audit failure.  The PR template mentioned a different audit command, which did not fail for me.

I did run the test earlier and it had succeeded (but with warnings).  For some reason, I can't run the test any more.
```
$ brew test global
Error: Testing requires the latest version of global

$ brew info global
global: stable 6.6.6, HEAD
Source code tag system
https://www.gnu.org/software/global/
/opt/homebrew/Cellar/global/6.6.6_1 (625 files, 8.2MB) *
  Built from source on 2021-05-24 at 01:11:55
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/global.rb
==> Dependencies
Required: ctags ✔, python@3.9 ✔
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 308 (30 days), 1,023 (90 days), 5,063 (365 days)
install-on-request: 309 (30 days), 1,023 (90 days), 4,995 (365 days)
build-error: 0 (30 days)
```

https://github.com/Homebrew/homebrew-core/pull/76558 should fix the tests.  Would it make sense to merge that in separately?

**Comment by SMillerDev** [comment]
> I did run the test earlier and it had succeeded (but with warnings). For some reason, I can't run the test any more.

You need to have the latest version installed too. For you that would probably require `brew reinstall -s global`

**Comment by khirbat** [comment]
> > I did run the test earlier and it had succeeded (but with warnings). For some reason, I can't run the test any more.
> 
> You need to have the latest version installed too. For you that would probably require `brew reinstall -s global`

Doesn't the `brew info global` output in my earlier comment show that I built and installed the latest version?

**Comment by SMillerDev** [comment]
No, that one has `revision 1` and the PR doesn't

**Review by carlocab — COMMENTED** [review]
Release announcement: https://lists.gnu.org/archive/html/info-global/2021-05/msg00000.html

Thanks for your first contribution to Homebrew/core, @khirbat.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**